mlflow/server/auth/__init__.py

```diff
@@ -585,6 +677,80 @@ def filter_search_experiments(resp: Response):
     resp.data = message_to_json(response_message)
 
 
+def filter_search_logged_models(resp: Response) -> None:
+    """
+    Filter out unreadable logged models from the search results.
+    """
+    from mlflow.utils.search_utils import SearchLoggedModelsPaginationToken as Token
+
+    if sender_is_admin():
+        return
+
+    response_proto = SearchLoggedModels.Response()
+    parse_dict(resp.json, response_proto)
+
+    # fetch permissions
+    username = authenticate_request().username
+    perms = store.list_experiment_permissions(username)
+    can_read = {p.experiment_id: get_permission(p.permission).can_read for p in perms}
+    default_can_read = get_permission(auth_config.default_permission).can_read
+
+    # Remove unreadable models
+    for m in list(response_proto.models):
+        if not can_read.get(m.info.experiment_id, default_can_read):
+            response_proto.models.remove(m)
+
+    request_proto = _get_request_message(SearchLoggedModels())
+    max_results = request_proto.max_results
+    # These parameters won't change in the loop
+    params = {
+        "experiment_ids": list(request_proto.experiment_ids),
+        "filter_string": request_proto.filter or None,
+        "order_by": (
+            [
+                {
+                    "field_name": ob.field_name,
+                    "ascending": ob.ascending,
+                    "dataset_name": ob.dataset_name,
+                    "dataset_digest": ob.dataset_digest,
```

Comment:
```suggestion
                    "dataset_name": ob.dataset_name or None,
                    "dataset_digest": ob.dataset_digest or None,
```
They could be empty?